src/audio/src/src_generic.c

```diff
@@ -60,23 +60,17 @@ static inline void fir_filter_generic(int32_t *rp, const void *cp, int32_t *wp0,
 		 * Qx.31.
 		 */
 		for (i = 0; i < n1; i++) {
-			y0 += (int64_t)(*coef) * (*data);
-			data++;
-			y1 += (int64_t)(*coef) * (*data);
-			data++;
-			coef++;
+			y0 += (int64_t)(*coef) * (*(data++));
+			y1 += (int64_t)(*(coef++)) * (*(data++));
 		}
 
 		/* No need to check for circular wrap. Pointer data is moved to
 		 * fir_start to be used by next loop if n2 is greater than zero.
 		 */
 		data = fir_start;
 		for (i = 0; i < n2; i++) {
```

Comment:
you'd achieve the same "code compactification" without the IMHO slightly confusing notation `*(x++)` by doing
```
for (i = 0; i < n1; i++, coef++, data += 2) {
	y0 += (int64_t)*coef * data[0];
	y1 += (int64_t)*coef * data[1];
}
```

Reply:
Yes, less lines but this version of code may do more operations than current patch?

Reply:
++ operation always followed with current data, no need separate, this is aligned with intrinsic code.